libraries/StorageManager/StorageManager.cpp

```diff
@@ -351,6 +351,7 @@ bool StorageAccess::attach_file(const char *filename, uint16_t size_kbyte)
     auto *newfile = new FileStorage;
     if (newfile == nullptr) {
         AP_BoardConfig::allocation_error("StorageFile");
+        return false;
```

Comment:
allocation_error is marked "NORETURN" - if we get ehere then something has*seriously*  gone wrong

https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_BoardConfig/AP_BoardConfig.h#L36